classes/page_group.php

```diff
@@ -54,6 +83,27 @@ protected function set_fuzzydurationcounts(array $data) {
         $this->raw_set('fuzzydurationcounts', json_encode($data, JSON_FORCE_OBJECT));
     }
 
+    /**
+     * Called after the page group is created in the DB.
+     */
+    public function after_create() {
+        $this->after_update(true);
+    }
+
+    /**
+     * Called after the DB is updated.
+     *
+     * @param bool $result
+     */
+    public function after_update($result) {
+        if ($result) {
+            $keydata = ['name' => $this->get('name'), 'month' => $this->get('month')];
+            $cachekey = serialize($keydata);
+            $cache = \cache::make('tool_excimer', self::CACHE_NAME);
+            $cache->set($cachekey, $this->to_record());
```

Comment:
Similar to an above comment, could we pull out this block into an internal method and call it when needed. I was going to ask for a comment to be added here but this seems like a better approach given the need to reuse it.

For example:
```
$this->update_cache();
```